mbsubmit: cleanup and completion

diego-plan9 opened the pull request:
Glad to see a new release has been made!

I'm getting back to work on beets after a few days away from the computer, hopefully bringing issue #1689 to a close eventually. As hinted on the previous discussion, this pull request is intended to take care of the `mbsubmit` plugin cleanup, now that the underlying pieces are in place.

I have modified a bit the behaviour, making the decision of appending the `"Print tracks"` choice depend solely on `task.rec`. The default behaviour is to only append the choice to matches where the recommendation is equal or lower than `Recommendation.medium`, which hopefully covers the most obvious choices (albums with no matches, albums with weak-ish matches) and the original request by @awesomer, and also avoids polluting the prompt in the cases where the match is strong. A config option has been added that allows the user to modify this settings (extra-picky users might find it useful to always be able to print tracks for fixing spelling mistakes, other users might only want it on albums with no matches, etc).

Other than that, a configuration option for setting the format string has been added as well - I can't think of a case where this might come in handy currently, but maybe more creative users might find it useful.

A couple of notes:
* currently, the plugin makes no effort of nicely formatting items that might be lacking some of the required fields. Would it be useful to add some extra checks and fall back to printing the filename (or something more advanced with the help of `fromfilename`, etc) in those cases?
* there might be some problems on some combination on options: for example, if the user sets the threshold to `strong`, but launches the importer in non-timid mode, the prompt will not actually be displayed. Would a note on the (upcoming) documentation suffice, as handling this case probably requires some changes that seem to be a bit out of the scope of the plugin?

As usual, any comments and input are more than welcome!

sampsyo commented on beetsplug/mbsubmit.py:
```diff
@@ -35,21 +36,26 @@ class MBSubmitPlugin(BeetsPlugin):
+            'threshold': 'medium',
+        })
+
+        # validate and store threshold
```
We tend to format comments as "full sentences" in beets—starting with a capital letter and ending with a period.

sampsyo commented on test/test_mbsubmit.py:
```diff
@@ -0,0 +1,72 @@
+        self.unload_plugins()
+        self.teardown_beets()
+
+    def test_xxxxprint_tracks_output(self):
```
Is there a reason for the `xxx` in the test names?

diego-plan9 replied:
Yes, there is a powerful reason: my sometimes absent mind! :hammer:

On a more serious note, I did notice just after submitting the pull request and already fixed it on my local copy. The current code was still meant to have a final cleanup phase on my end, that will also address the issue with the "non-full sentence" comments - sorry about that.

sampsyo commented:
Awesome; this looks perfect. Feel free to merge at will.

On those questions:

- I'd say we shouldn't worry about missing fields. Here's my reasoning: this plugin is about situations where you already have good metadata (or better metadata than MB). So while picking up tags from filenames à la `fromfilename` would be nifty, it's out of scope for this initial version.
- I see what you're saying—yeah, let's document this.

diego-plan9 commented:
Thanks for the quick turnaround, and commenting on the questions - sounds like a reasonable compromise. I will work on the documentation during tomorrow and hopefully everything should be ready for a merge soon!

diego-plan9 commented:
Fixed the style issues, and added some pretty basic documentation and a changelog entry. As usual, double checking the documentation would be more than welcome!

sampsyo commented on docs/plugins/mbsubmit.rst:
```diff
@@ -0,0 +1,54 @@
+As MusicBrainz currently does not support submitting albums programmatically,
+the recommended workflow is to copy the output of the ``Print tracks`` choice
+and paste it into the parser that can be found by clicking on the
+"`Track Parser`" button on MusicBrainz "`Tracklist`" tab.
```
Any particular reason for the extra backticks here? It seems like quotes might be enough. (This is officially the tiniest style issue ever!)

sampsyo commented on docs/plugins/mbsubmit.rst:
```diff
@@ -0,0 +1,54 @@
+To configure the plugin, make a ``mbsubmit:`` section in your configuration
+file. The following options are available:
+
+- **mb_format**: The format used for printing the tracks, defined using the
```
Maybe just `format` would be OK? Since this is an `mbsubmit` plugin, the `mb_` prefix could be seen as redundant.

sampsyo commented on docs/plugins/mbsubmit.rst:
```diff
@@ -0,0 +1,54 @@
+require other ``beets`` command line switches to be enabled in order to work as
+intended. In particular, setting a threshold of ``strong`` will only display
+the prompt if ``timid`` mode is enabled. You can find more information about
+how the recommendation system works on :ref:`match-config`.
```
"on" -> "at" or "in"

sampsyo commented:
This looks great! I made some extremely minor suggestions in the docs, but I think this is ready—feel free to merge this whenever you're happy with it.

Woohoo! :tada:

diego-plan9 commented:
Fixed the style issues (the extra ticks were for making it more evident that they were referring to the names of the button/elements on MusicBrainz site, and the choice of `mb_format` for the configuration was to avoid any confusion with the `format` flag - but I'm happy to drop both)!

I have also merged upstream/master, solving the tiny conflict on `changelog.rst`.

diego-plan9 commented:
And merged, after the checks seemed ok!

sampsyo commented:
Woohoo!
